### PR TITLE
[SYCL][ROCm] Setup lit tests for ROCm plugin

### PR DESCRIPTION
**Pull request by npmiller**
This patch wires up the lit tests for the ROCm plugin adding `check-sycl-rocm`.

`check-sycl-rocm` includes `check-sycl-rocm-on-device` and `check-sycl-rocm-ptx` to run the compiler tests with the NVIDIA triple if using the NVIDIA platform for ROCm or `check-sycl-rocm-gcn` if using the AMD platform.

This PR is marked as draft is it requires a bit more work to get `check-sycl-rocm` to work properly.

On AMD it should pass after these two patches are merged/fixed:
* https://github.com/intel/llvm/pull/4139
* https://github.com/intel/llvm/pull/4142

On NVIDIA it requires a bit more test filtering and/or debugging because of issues in the HIP APIs, https://github.com/intel/llvm/pull/4123 is one of them but there is a few other bugs that need to be addressed or filtered before this PR can be merged.

### CODE REVIEW

File: sycl/test/CMakeLists.txt
Comment (bader) [inline sycl/test/CMakeLists.txt:82]: Very minor comment. 

See the discussion in https://github.com/intel/llvm/pull/4069. In case we need to support other spellings for `SYCL_BUILD_PI_ROCM_PLATFORM`, we might need to make sure it's all capital.

**Review by bader — COMMENTED** [review]
LGTM. Thanks!

**Comment by npmiller** [comment]
I've marked the remaining two failing tests with NVIDIA as XFAIL, so `check-sycl-rocm` should now pass on NVIDIA platform with this branch.

File: sycl/test/CMakeLists.txt
Comment (npmiller) [reply sycl/test/CMakeLists.txt:82]: Yeah that's probably a good idea, however there's already a bunch of other places that do this comparison. So I'd suggest to keep it like that in this patch and then once it's merged I can put up a new PR updating all of the uses of this to be case-insensitive.

**Comment by bader** [comment]
@npmiller, could you resolve merge conflicts, please? It looks like a number of tests were just removed - https://github.com/intel/llvm/commit/7735139bb4d8420fcde692b6eddd2ebf9257efe8.

**Comment by npmiller** [comment]
> @npmiller, could you resolve merge conflicts, please? It looks like a number of tests were just removed - [7735139](https://github.com/intel/llvm/commit/7735139bb4d8420fcde692b6eddd2ebf9257efe8).

Rebased and all conflicts resolved

**Review by bader — DISMISSED** [review]
LGTM, although there is some contradiction between this and another your patch - https://github.com/intel/llvm/pull/4239. I think they suggest using different options for setting target architecture.

**Comment by npmiller** [comment]
> LGTM, although there is some contradiction between this and another your patch - #4239. I think they suggest using different options for setting target architecture.

Yes there is, because #4239 changes the option that should be used. The PRs are independent at the moment so either one can go in first but I'll have to update the other to resolve conflict and/or update the new changes.

**Review by vladimirlaz — DISMISSED** [review]

**Comment by bader** [comment]
@DenisBakhvalov, @kbobrovs, @kychendev, ping.

**Review by DenisBakhvalov — DISMISSED** [review]
Changes in `sycl/test/esimd/odr.cpp` look good.

**Comment by kychendev** [comment]
> Changes in `sycl/test/esimd/odr.cpp` look good.

LGTM

**Comment by bader** [comment]
> On NVIDIA it requires a bit more test filtering and/or debugging because of issues in the HIP APIs, #4123 is one of them but there is a few other bugs that need to be addressed or filtered before this PR can be merged.

We are not doing ROCm plug-in validation on NVIDIA cards, so I'm not sure if it make sense to block the merge of this pull request by #4123. 
@npmiller, are you okay to merge this PR before #4123?

**Comment by npmiller** [comment]
> > On NVIDIA it requires a bit more test filtering and/or debugging because of issues in the HIP APIs, #4123 is one of them but there is a few other bugs that need to be addressed or filtered before this PR can be merged.
> 
> We are not doing ROCm plug-in validation on NVIDIA cards, so I'm not sure if it make sense to block the merge of this pull request by #4123.
> @npmiller, are you okay to merge this PR before #4123?

Yes, the tests addressed in #4123 are marked as XFAIL here, so this MR should pass `check-sycl-rocm` on NVidia too at the moment.

I do need to change the `-mcpu` to `--offload-arch` now that #4239 is merged.
I'll have a patch to do that tomorrow. If that's easier to review we could merge this and have that patch in a follow up PR, this will mostly work without it, lit will just be a bit annoying as it looks for `-mcpu` but specifying both flags should work.

**Comment by bader** [comment]
> I do need to change the `-mcpu` to `--offload-arch` now that #4239 is merged.
> I'll have a patch to do that tomorrow. If that's easier to review we could merge this and have that patch in a follow up PR, this will mostly work without it, lit will just be a bit annoying as it looks for `-mcpu` but specifying both flags should work.

If merging this patch is not urgent, let's make this change here. There should be no difference in where such change is reviewed - within this PR or separately.

**Comment by npmiller** [comment]
Rebased and updated the uses of `-mcpu` to use the new `--offload-arch`

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@v-klochkov, it looks like failure in reduction test on Windows is not related to the patch. Please, take a look.
@pvchupin, FYI.

**Review by vladimirlaz — APPROVED** [review]

**Comment by v-klochkov** [comment]
> @v-klochkov, it looks like failure in reduction test on Windows is not related to the patch. Please, take a look.
> @pvchupin, FYI.
Fix for timeout on ACC:
https://github.com/intel/llvm-test-suite/pull/414

**Pull request merged**
